codes/models/CHN_generate_data_and_model_projection.R

```diff
@@ -14,11 +17,12 @@ mydata <- read_csv('models/reg_data/CHN_reg_data.csv',
                      date = col_date(format = ""),
                      t = col_character(),
                      adm2_id = col_character(),
-                     adm1_id = col_character()
+                     adm1_id = col_character(),
+                     adm1_adm2_name = col_character(),
+                     day_avg = col_double()
                    )) %>% 
   arrange(adm1_name, adm2_name, date) %>%
-  mutate(adm12_id = factor(adm12_id)) %>%
-  mutate(tmp_id = factor(adm12_id))
+  mutate(tmp_id = factor(adm1_adm2_name))
 
 changed = TRUE
```

Comment:
@kendonB why the `while` loop here?  Seems like the first time through the loop is the only time anything could possibly change. And if the `nrow` condition is not met after the first iteration, then I think the loop will just be stuck forever (??) .   If this is right, maybe a `stop()` call would be better?  Or am I missing something?

Reply:
The loop will run after each time the `new` df is different to `mydata`. The loop deletes rows with leading missing values such that for each unit the first date has non missing data. So it's going to keep running until it stops deleting rows. Let me know if you think this doesn't work

Reply:
I'm gonna trust you on this one Kenny!

Reply:
nevermind i get it now.  (nix the trusting comment :)  )